src/Util/Amiibo.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <basis/seadTypes.h>
+
+namespace al {
+struct NfpCharacterId;
+struct NfpInfo;
+class IUseMessageSystem;
+class IUseSceneObjHolder;
+}  // namespace al
+
+class ShopItem {
+public:
+    struct ItemInfo;
+};
```

Comment:
Is this the correct location for this class? If not, move it somewhere else and `include` - it might cause issues later, as you can't define a class multiple times.

Reply:
I cannot find a proper place for this. When somebody takes a harder look at it then it'll be a better chance to place it somewhere. I also want to keep files that only include a forward decl (like this would be) to a minimum.

Reply:
It's not only a stylistic choice, but will be a compile problem later on if not done correctly... I'd say this is important then.

Reply:
Changed to namespace due to all uses being structs inside of it- this allows for it to be forward decl'd without any issue